src/client/batch.rs

```diff
@@ -1,191 +1,106 @@
+    let balance_info = galoy_cli.fetch_balance(Some(wallet_type), Vec::new())?;
+    let current_balance: Decimal = balance_info.iter().map(|info| info.balance).sum();
+
+    let mut total_payment_amount: Decimal = Decimal::new(0, 0);
+    for record in records {
+        let amount: Decimal = Decimal::from_str(record.get(1).unwrap_or_default())?;
+        total_payment_amount += amount;
     }
 
-    pub fn check_limit(&self) -> anyhow::Result<()> {
-        todo!("Check limit. need API on the backend for it");
+    if total_payment_amount > current_balance {
+        return Err(anyhow::anyhow!("Insufficient balance in the wallet"));
     }
 
-    pub fn check_balance(&self) -> anyhow::Result<()> {
-        let me = self.client.me()?;
-        let me_wallet_id = me.default_account.default_wallet_id;
-
-        let mut total_sats = dec!(0);
+    Ok(())
+}
 
-        for payment in self.payments.iter() {
-            let sats = match payment.sats {
-                Some(value) => value,
-                None => bail!("sats needs to be populated first"),
-            };
-            total_sats += sats;
-        }
+pub fn execute_batch_payment(
+    records: &[csv::StringRecord],
+    wallet_type: Wallet,
+    galoy_cli: &GaloyClient,
+) -> anyhow::Result<()> {
+    for record in records {
+        let username = record
+            .get(0)
+            .ok_or(anyhow::anyhow!("Username is missing"))?;
 
-        let me_default_wallet = me
-            .default_account
-            .wallets
-            .iter()
-            .find(|wallet| wallet.id == me_wallet_id);
-
-        let balance_sats = match me_default_wallet {
-            Some(value) => value.balance,
-            None => bail!("no balance"),
-        };
-        if total_sats > balance_sats {
-            bail!(
-                "not enough balance, got {}, need {}",
-                balance_sats,
-                total_sats
-            )
-        }
+        let amount: Decimal = Decimal::from_str(record.get(1).unwrap_or_default())?;
 
-        Ok(())
-    }
+        let memo = record.get(2).map(|s| s.to_string());
 
-    pub fn show(&self) {
-        println!("{:#?}", &self.payments)
-    }
-
-    pub fn execute(&mut self) -> anyhow::Result<()> {
-        self.check_self_payment()?;
-        self.check_balance()?;
-
-        for Payment {
-            username,
-            memo,
-            usd,
-            sats,
-            ..
-        } in self.payments.drain(..)
-        {
-            let amount = match sats {
-                Some(value) => value,
-                None => bail!("need sats amount"),
-            };
-            let res = &self
-                .client
-                .intraleger_send(username.clone(), amount, memo)
-                .context("issue sending intraledger")?;
-
-            println!(
-                "payment to {username} of sats {amount}, usd {usd}: {:?}",
-                res
-            );
+        match wallet_type {
```

Comment:
How are you dealing with partial failures? What if the first 5 out of 10 payments complete but then there is an error - how is this reported to avoid a double payment on re-try?